Runtime/FilesManagement/Protocols/UnityAudioProtocol.cs

```diff
@@ -33,38 +42,54 @@ public async Task<object> Read<T>(string uri)
         
 			var requestUri = webRequestProtocol + path;
         
-			var uwr = UnityWebRequestMultimedia.GetAudioClip(requestUri, AudioType.MPEG);
-        
-			((DownloadHandlerAudioClip)uwr.downloadHandler).streamAudio = streamAudio;
+			var dh = new DownloadHandlerAudioClip(requestUri, AudioType.MPEG);
+			dh.compressed = true; // This
  
-			await uwr.SendWebRequest().AsTask(coroutineProxy);
-        
-			if (uwr.isNetworkError || uwr.isHttpError)
+			using (UnityWebRequest wr = new UnityWebRequest(requestUri, "GET", dh, null)) 
 			{
-				throw new FileNotFoundException($"File not found for uri: {requestUri}");
-			}
-            
-			var dlHandler = (DownloadHandlerAudioClip)uwr.downloadHandler;
- 
-			if (dlHandler.isDone)
-			{
-				var clip = DownloadHandlerAudioClip.GetContent(uwr);
-				clip.name = fileNameGenerator.GetName(uri);
-				return clip;
+				await wr.SendWebRequest().AsTask(coroutineProxy);
+				if (wr.responseCode == 200) 
+				{
+					return dh.audioClip;
+				}
+				Debug.LogError($"Download failed. Uri {requestUri} Response {wr.responseCode}. Error: {wr.error}");
 			}
         
 			Debug.LogError("The download process is not completely finished.");
 			return null;
 		}
 
-		public Task<File<T>> Write<T>(string uri, T value)
+		public async Task<File<T>> Write<T>(string uri, T value)
 		{
-			throw new System.NotImplementedException();
+			var path = GetFullPath(uri);
```

Comment:
This should definitely not do this, you should throw a NotImplementedException instead